src/Microsoft.AspNetCore.Mvc.Modules/SatalliteApplicationPart.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Modules;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.Extensions.DependencyInjection;
+using Orchard.Environment.Shell.Builders.Models;
+
+namespace Microsoft.AspNetCore.Mvc.Modules
+{
+    /// <summary>
+    /// An <see cref="ApplicationPart"/> backed by an <see cref="Assembly"/>.
+    /// </summary>
+    public class SatalliteApplicationPart :
+        ApplicationPart,
+        IApplicationPartTypeProvider,
+        ICompilationReferencesProvider
+    {
+        /// <summary>
+        /// Initalizes a new <see cref="AssemblyPart"/> instance.
+        /// </summary>
+        /// <param name="assembly"></param>
+        public SatalliteApplicationPart(IHttpContextAccessor httpContextAccessor)
```

Comment:
I cant, the part gets initialized within a Startup class that doesnt have access to the ShellBlueprint.